p2p/src/cybershake.rs

```diff
@@ -46,11 +46,16 @@ use merlin::Transcript; // TODO: change for raw Strobe.
 use tokio::io;
 use tokio::prelude::*;
 
+use futures::task::{Context, Poll};
 use serde::{Deserialize, Serialize};
+use std::io::{Read, Write};
+use std::pin::Pin;
 
 /// The current version of the protocol is 0.
 /// In the future we may add more versions, version bits or whatever.
 const ONLY_SUPPORTED_VERSION: u64 = 0;
+const PLAINTEXT_BUF_SIZE: u16 = 4096;
```

Comment:
This constant should contain "max" in the name since the actual chunk could be smaller when flushed.
```suggestion
/// This is the maximum amount of bytes that we are collecting
/// before encrypting and sending into the underlying socket.
/// This is not a limit imposed by the receiver.
/// We can receive chunks up to 2**16-1 bytes long (where the length fits in u16).
const PLAINTEXT_MAX_BUF_SIZE: u16 = 4096;
```

Reply:
But this constant mean the size of the buffer, not mean size of chunks.

Reply:
fair.